performance-test/src/puppeteer-test.ts

```diff
@@ -102,7 +102,77 @@ export const setupBrowser = async (): Promise<{
 
 const ResizeButtonXPath = "//a[contains(., 'P R')]"
 
-async function initialiseProject(page: puppeteer.Page): Promise<void> {
+interface Baselines {
+  basicCalc: FrameResult
+  simpleDispatch: FrameResult
+}
+
+function calculatePi(accuracy: number): number {
+  // Uses the Nilakantha series
+  let i = 2
+  let j = 3
+  let k = 4
+  let count = 0
+  let result = 3
+
+  while (count < accuracy) {
+    const nextAdjustment = 4 / (i * j * k)
+    if (count % 2 === 0) {
+      result += nextAdjustment
+    } else {
+      result -= nextAdjustment
+    }
+
+    i += 2
+    j += 2
+    k += 2
+
+    count++
+  }
+
+  return result
+}
+
+function timeBasicCalc(): FrameResult {
+  let times: Array<number> = []
+
+  for (let i = 0; i < 100; i++) {
+    const start = Date.now()
+    calculatePi(7500000)
+    const end = Date.now()
+    times.push(end - start)
+  }
+
+  const sortedTimes = times.sort((a, b) => a - b)
+  const totalTime = sortedTimes.reduce((sum, next) => sum + next, 0)
+
+  const analytics = {
+    frameMin: sortedTimes[0]!,
+    frameMax: sortedTimes[sortedTimes.length - 1]!,
+    frameAvg: Number((totalTime / sortedTimes.length).toFixed()),
+    percentile25: sortedTimes[Math.floor(sortedTimes.length * 0.25)],
+    percentile50: sortedTimes[Math.floor(sortedTimes.length * 0.5)],
+    percentile75: sortedTimes[Math.floor(sortedTimes.length * 0.75)],
+  }
+  return {
+    title: 'Calc Pi 5m',
```

Comment:
what does 5m mean ?

Reply:
(5m iterations, not 5m digits)

Reply:
> (5m iterations, not 5m digits)

massively disappointing!

Reply:
Look, I know what you're saying, and that I bring shame on my family, but even [NASA only uses pi correct to 15dp](https://www.jpl.nasa.gov/edu/news/2016/3/16/how-many-decimals-of-pi-do-we-really-need/):
> In other words, by cutting pi off at the 15th decimal point, we would calculate a circumference for that circle that is very slightly off. It turns out that our calculated circumference of the 25 billion mile diameter circle would be wrong by 1.5 inches. Think about that.

Think about that Balazs.